--- src/libappindicator/src/app-indicator.c	2015-02-05 02:06:40.061527884 +0900
+++ src/libappindicator/src/app-indicator.c	2015-02-05 02:06:53.914417292 +0900
@@ -50,18 +50,18 @@
 
 #define PANEL_ICON_SUFFIX  "panel"
 
-/**
+/*
  * AppIndicatorPrivate:
  * All of the private data in an instance of an application indicator.
  *
  * Private Fields
- * @id: The ID of the indicator.  Maps to AppIndicator:id.
- * @category: Which category the indicator is.  Maps to AppIndicator:category.
- * @status: The status of the indicator.  Maps to AppIndicator:status.
- * @icon_name: The name of the icon to use.  Maps to AppIndicator:icon-name.
- * @attention_icon_name: The name of the attention icon to use.  Maps to AppIndicator:attention-icon-name.
- * @menu: The menu for this indicator.  Maps to AppIndicator:menu
- * @watcher_proxy: The proxy connection to the watcher we're connected to.  If we're not connected to one this will be %NULL.
+ * id: The ID of the indicator.  Maps to AppIndicator:id.
+ * category: Which category the indicator is.  Maps to AppIndicator:category.
+ * status: The status of the indicator.  Maps to AppIndicator:status.
+ * icon_name: The name of the icon to use.  Maps to AppIndicator:icon-name.
+ * attention_icon_name: The name of the attention icon to use.  Maps to AppIndicator:attention-icon-name.
+ * menu: The menu for this indicator.  Maps to AppIndicator:menu
+ * watcher_proxy: The proxy connection to the watcher we're connected to.  If we're not connected to one this will be %NULL.
  */
 struct _AppIndicatorPrivate {
 	/*< Private >*/
