--- src/os-patch-indicator-session-trusty/src/backend-dbus/actions.c~	2014-07-23 13:35:16.027357725 +0900
+++ src/os-patch-indicator-session-trusty/src/backend-dbus/actions.c	2014-08-10 03:28:25.135881546 +0900
@@ -100,14 +100,14 @@
   if (!g_settings_get_boolean (p->indicator_settings, "suppress-logout-restart-shutdown"))
     {
       /* can we use the Unity prompt? */
-      if ((prompt == PROMPT_NONE) && p && p->end_session_dialog)
+      /*if ((prompt == PROMPT_NONE) && p && p->end_session_dialog)
         {
           GDBusProxy * proxy = G_DBUS_PROXY (p->end_session_dialog);
           char * name = g_dbus_proxy_get_name_owner (proxy);
           if (name != NULL)
             prompt = PROMPT_WITH_UNITY;
           g_free (name);
-        }
+        }*/
 
       /* can we use zenity? */
       if ((prompt == PROMPT_NONE) && p && p->zenity)
