--- src/gala/src/Main.vala	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/src/Main.vala.qq	2015-10-31 01:11:23.554291511 +0900
@@ -40,7 +40,7 @@
 
 		Meta.Plugin.manager_set_plugin_type (new WindowManagerGala ().get_type ());
 
-		Meta.set_wm_name ("Mutter(Gala)");
+		//Meta.set_wm_name ("Mutter(Gala)");
 
 		/**
 		 * Prevent Meta.init () from causing gtk to load gail and at-bridge
--- src/gala/src/Makefile.am	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/src/Makefile.am.qq	2015-10-31 01:45:47.751708287 +0900
@@ -33,56 +33,12 @@
 gala_VALASOURCES = \
 	DBus.vala \
 	DockThemeManager.vala \
-	DragDropAction.vala \
-	InternalUtils.vala \
 	Main.vala \
 	PluginManager.vala \
 	ScreenSaver.vala \
 	Settings.vala \
-	ShadowEffect.vala \
-	TextShadowEffect.vala \
-	WindowListener.vala \
-	WindowManager.vala \
-	WorkspaceManager.vala \
-	Widgets/IconGroup.vala \
-	Widgets/IconGroupContainer.vala \
-	Widgets/MonitorClone.vala \
-	Widgets/MultitaskingView.vala \
-	Widgets/SafeWindowClone.vala \
-	Widgets/WindowClone.vala \
-	Widgets/WindowCloneContainer.vala \
-	Widgets/WindowIconActor.vala \
-	Widgets/WindowMenu.vala \
-	Widgets/WindowOverview.vala \
-	Widgets/WindowSwitcher.vala \
-	Widgets/WorkspaceClone.vala \
-	Widgets/WorkspaceInsertThumb.vala \
 	$(NULL)
 
-gala_background = \
-	Background/Animation.vala \
-	Background/Background.vala \
-	Background/BackgroundCache.vala \
-	Background/BackgroundContainer.vala \
-	Background/BackgroundManager.vala \
-	Background/BackgroundSource.vala \
-	Background/SystemBackground.vala \
-	$(NULL)
-
-gala_background312 = \
-	Background-3.12/Background.vala \
-	Background-3.12/BackgroundCache.vala \
-	Background-3.12/BackgroundManager.vala \
-	Background-3.12/SlideShow.vala \
-	Background-3.12/SystemBackground.vala \
-	$(NULL)
-
-if HAVE_MUTTER314
-gala_VALASOURCES += $(gala_background)
-else
-gala_VALASOURCES += $(gala_background312)
-endif
-
 nodist_gala_SOURCES = \
 	gala_vala.stamp \
 	$(gala_VALASOURCES:.vala=.c) \
@@ -102,8 +58,6 @@
 	$(NULL)
 
 EXTRA_DIST = \
-	$(gala_background) \
-	$(gala_background312) \
 	$(gala_VALASOURCES) \
 	$(NULL)
 
--- src/gala/src/PluginManager.vala	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/src/PluginManager.vala.qq	2015-10-31 01:40:34.861022060 +0900
@@ -30,21 +30,21 @@
 			return instance;
 		}
 
-		public signal void regions_changed ();
+		//public signal void regions_changed ();
 
 		public bool initialized { get; private set; default = false; }
 
-		public X.Xrectangle[] regions { get; private set; }
+		//public X.Xrectangle[] regions { get; private set; }
 
-		public string? window_switcher_provider { get; private set; default = null; }
-		public string? desktop_provider { get; private set; default = null; }
-		public string? window_overview_provider { get; private set; default = null; }
-		public string? workspace_view_provider { get; private set; default = null; }
+		//public string? window_switcher_provider { get; private set; default = null; }
+		//public string? desktop_provider { get; private set; default = null; }
+		//public string? window_overview_provider { get; private set; default = null; }
+		//public string? workspace_view_provider { get; private set; default = null; }
 
 		HashTable<string,Plugin> plugins;
 		File plugin_dir;
 
-		WindowManager? wm = null;
+		//WindowManager? wm = null;
 
 		Gee.LinkedList<PluginInfo?> load_later_plugins;
 
@@ -133,17 +133,17 @@
 
 			if (initialized) {
 				initialize_plugin (info.module_name, plugin);
-				recalculate_regions ();
+				//recalculate_regions ();
 			}
 		}
 
 		void initialize_plugin (string plugin_name, Plugin plugin)
 		{
-			plugin.initialize (wm);
-			plugin.region_changed.connect (recalculate_regions);
+			plugin.initialize ();
+			//plugin.region_changed.connect (recalculate_regions);
 		}
 
-		bool check_provides (string name, PluginFunction provides)
+		/* bool check_provides (string name, PluginFunction provides)
 		{
 			var message = "Plugins %s and %s both provide %s functionality, using first one only";
 			switch (provides) {
@@ -178,14 +178,14 @@
 			}
 
 			return true;
-		}
+		} */
 
-		public void initialize (WindowManager _wm)
+		public void initialize ()
 		{
-			wm = _wm;
+			//wm = _wm;
 
 			plugins.@foreach (initialize_plugin);
-			recalculate_regions ();
+			//recalculate_regions ();
 
 			initialized = true;
 		}
@@ -208,7 +208,7 @@
 		 * Iterate over all plugins and grab their regions, update the regions
 		 * array accordingly and emit the regions_changed signal.
 		 */
-		void recalculate_regions ()
+		/* void recalculate_regions ()
 		{
 			X.Xrectangle[] regions = {};
 
@@ -227,7 +227,7 @@
 
 			this.regions = regions;
 			regions_changed ();
-		}
+		} */
 	}
 }
 
--- src/gala/data/Makefile.am	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/data/Makefile.am.qq	2015-10-31 01:44:05.684420301 +0900
@@ -6,7 +6,7 @@
 styles_DATA = gala.css texture.png close.png
 
 applicationsdir = $(datadir)/applications
-applications_DATA = gala.desktop gala-wayland.desktop gala-multitaskingview.desktop
+applications_DATA = gala.desktop gala-wayland.desktop
 
 @INTLTOOL_DESKTOP_RULE@
 
@@ -25,14 +25,12 @@
 	gala.css \
 	gala.desktop \
 	gala-wayland.desktop \
-	gala-multitaskingview.desktop.in \
 	texture.png \
 	close.png \
 	org.pantheon.desktop.gala.gschema.xml.in.in \
 	$(NULL)
 
 CLEANFILES = \
-	gala-multitaskingview.desktop \
 	gschemas.compiled \
 	org.pantheon.desktop.gala.gschema.xml \
 	org.pantheon.desktop.gala.gschema.xml.in \
--- src/gala/lib/Makefile.am	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/lib/Makefile.am.qq	2015-10-31 02:02:23.186258738 +0900
@@ -47,11 +47,8 @@
 	$(NULL)
 
 libgala_la_VALASOURCES = \
-	ActivatableComponent.vala \
 	Plugin.vala \
 	Utils.vala \
-	WindowIcon.vala \
-	WindowManager.vala \
 	$(NULL)
 
 nodist_libgala_la_SOURCES = \
--- src/gala/lib/Plugin.vala	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/lib/Plugin.vala.qq	2015-10-31 02:22:37.651250838 +0900
@@ -19,11 +19,7 @@
 {
 	public enum PluginFunction
 	{
-		ADDITION,
-		WINDOW_SWITCHER,
-		DESKTOP,
-		WORKSPACE_VIEW,
-		WINDOW_OVERVIEW
+		ADDITION
 	}
 
 	public enum LoadPriority
@@ -82,7 +78,7 @@
 		/**
 		 * Emitted when update_region is called. Mainly for internal purposes.
 		 */
-		public signal void region_changed ();
+		//public signal void region_changed ();
 
 		/**
 		 * The region indicates an area where mouse events should be sent to
@@ -91,14 +87,14 @@
 		 * It is calculated by the system whenever update_region is called.
 		 * You can influce it with the custom_region and the track_actor function.
 		 */
-		public Meta.Rectangle[] region { get; private set; }
+		//public Meta.Rectangle[] region { get; private set; }
 
 		/**
 		 * This list will be merged with the region property. See region for
 		 * more details. Changing this property will cause update_region to be
 		 * called. Default to null.
 		 */
-		protected Meta.Rectangle[]? custom_region {
+		/* protected Meta.Rectangle[]? custom_region {
 			get {
 			   return _custom_region;
 			}
@@ -106,14 +102,14 @@
 				_custom_region = value;
 				update_region ();
 			}
-		}
+		} */
 
 		/**
 		 * Set this property to true while animating an actor if you have tracked
 		 * actors to prevent constant recalculations of the regions during an
 		 * animation.
 		 */
-		protected bool freeze_track {
+		/* protected bool freeze_track {
 			get {
 				return _freeze_track;
 			}
@@ -123,11 +119,11 @@
 				if (!_freeze_track)
 					update_region ();
 			}
-		}
+		} */
 
-		private bool _freeze_track = false;
-		private Meta.Rectangle[]? _custom_region = null;
-		private List<Clutter.Actor> tracked_actors = new List<Clutter.Actor> ();
+		//private bool _freeze_track = false;
+		//private Meta.Rectangle[]? _custom_region = null;
+		//private List<Clutter.Actor> tracked_actors = new List<Clutter.Actor> ();
 
 		/**
 		 * Once this method is called you can start adding actors to the stage
@@ -135,7 +131,7 @@
 		 *
 		 * @param wm The window manager.
 		 */
-		public abstract void initialize (WindowManager wm);
+		public abstract void initialize ();
 
 		/**
 		 * This method is currently not called in the code, however you should
@@ -155,9 +151,9 @@
 		public void track_actor (Clutter.Actor actor)
 		{
 			tracked_actors.prepend (actor);
-			actor.allocation_changed.connect (actor_allocation_changed);
+			//actor.allocation_changed.connect (actor_allocation_changed);
 
-			update_region ();
+			//update_region ();
 		}
 
 		/**
@@ -169,8 +165,8 @@
 		public void untrack_actor (Clutter.Actor actor)
 		{
 			tracked_actors.remove (actor);
-			actor.allocation_changed.disconnect (actor_allocation_changed);
-		}
+			//actor.allocation_changed.disconnect (actor_allocation_changed);
+		} 
 
 		/**
 		 * You can call this method to force the system to update the region that
@@ -180,7 +176,7 @@
 		 * after setting freeze_track back to false after an animation to make the
 		 * wm aware of the new position of the actor in question.
 		 */
-		public void update_region ()
+		/* public void update_region ()
 		{
 			var has_custom = custom_region != null;
 			var len = tracked_actors.length () + (has_custom ? custom_region.length : 0);
@@ -208,13 +204,13 @@
 			region = regions;
 
 			region_changed ();
-		}
+		} */
 
-		private void actor_allocation_changed (Clutter.ActorBox box, Clutter.AllocationFlags f)
+		/* private void actor_allocation_changed (Clutter.ActorBox box, Clutter.AllocationFlags f)
 		{
 			if (!freeze_track)
 				update_region ();
-		}
+		} */
 	}
 }
 
--- src/gala/plugins/Makefile.am	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/plugins/Makefile.am.qq	2015-10-31 01:42:43.723598071 +0900
@@ -1,6 +1,4 @@
 SUBDIRS = \
-	maskcorners \
 	notify \
-	zoom \
 	$(NULL)
 
--- src/gala/plugins/notify/Main.vala	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/plugins/notify/Main.vala.qq	2015-10-31 02:21:56.355348105 +0900
@@ -22,29 +22,29 @@
 {
 	public class Main : Gala.Plugin
 	{
-		Gala.WindowManager? wm = null;
+		//Gala.WindowManager? wm = null;
 
 		NotifyServer server;
 		NotificationStack stack;
 
-		public override void initialize (Gala.WindowManager wm)
+		public override void initialize ()
 		{
-			this.wm = wm;
-			var screen = wm.get_screen ();
+			//this.wm = wm;
+			//var screen = wm.get_screen ();
 
-			stack = new NotificationStack (wm.get_screen ());
-			wm.ui_group.add_child (stack);
+			stack = new NotificationStack ();
+			//wm.ui_group.add_child (stack);
 			track_actor (stack);
 
-			stack.animations_changed.connect ((running) => {
+			/* stack.animations_changed.connect ((running) => {
 				freeze_track = running;
-			});
+			}); */
 
 			server = new NotifyServer (stack);
 
-			update_position ();
-			screen.monitors_changed.connect (update_position);
-			screen.workareas_changed.connect (update_position);
+			//update_position ();
+			//screen.monitors_changed.connect (update_position);
+			//screen.workareas_changed.connect (update_position);
 
 			Bus.own_name (BusType.SESSION, "org.freedesktop.Notifications", BusNameOwnerFlags.NONE,
 				(connection) => {
@@ -62,7 +62,7 @@
 				});
 		}
 
-		void update_position ()
+		/* void update_position ()
 		{
 			var screen = wm.get_screen ();
 			var primary = screen.get_primary_monitor ();
@@ -70,12 +70,12 @@
 
 			stack.x = area.x + area.width - stack.width;
 			stack.y = area.y;
-		}
+		} */
 
 		public override void destroy ()
 		{
-			if (wm == null)
-				return;
+			//if (wm == null)
+				//return;
 
 			untrack_actor (stack);
 			stack.destroy ();
--- src/gala/plugins/notify/NormalNotification.vala	2015-10-31 00:53:41.495647000 +0900
+++ src/gala/plugins/notify/NormalNotification.vala.qq	2015-10-31 02:30:26.125803067 +0900
@@ -284,7 +284,7 @@
 
 			// if no default action has been set, we fallback to trying to find a window for the
 			// notification's sender process
-			var window = get_window ();
+			/* var window = get_window ();
 			if (window != null) {
 				var workspace = window.get_workspace ();
 				var time = screen.get_display ().get_current_time ();
@@ -295,10 +295,10 @@
 					window.activate (time);
 
 				dismiss ();
-			}
+			} */
 		}
 
-		Window? get_window ()
+		/* Window? get_window ()
 		{
 			if (sender_pid == 0)
 				return null;
@@ -314,7 +314,7 @@
 			}
 
 			return null;
-		}
+		} */
 
 		void dismiss ()
 		{
