--- src/gala/lib/Plugin.vala	2015-11-03 03:06:41.938071061 +0900
+++ src/gala/lib/Plugin.vala.qq	2015-11-03 03:10:12.503760716 +0900
@@ -150,7 +150,7 @@
 		 */
 		public void track_actor (Clutter.Actor actor)
 		{
-			tracked_actors.prepend (actor);
+			//tracked_actors.prepend (actor);
 			//actor.allocation_changed.connect (actor_allocation_changed);
 
 			//update_region ();
@@ -164,7 +164,7 @@
 		 */
 		public void untrack_actor (Clutter.Actor actor)
 		{
-			tracked_actors.remove (actor);
+			//tracked_actors.remove (actor);
 			//actor.allocation_changed.disconnect (actor_allocation_changed);
 		} 
 
--- src/gala/plugins/notify/NotificationStack.vala	2015-11-06 23:46:42.462758000 +0900
+++ src/gala/plugins/notify/NotificationStack.vala.qq	2015-11-06 23:50:44.054955832 +0900
@@ -33,9 +33,9 @@
 
 		int animation_counter = 0;
 
-		public NotificationStack (Screen screen)
+		public NotificationStack ()
 		{
-			Object (screen: screen);
+			Object ();
 		}
 
 		construct

--- src/gala/src/Settings.vala	2015-11-07 00:06:54.898629000 +0900
+++ src/gala/src/Settings.vala.qq	2015-11-07 02:41:32.388206630 +0900
@@ -27,13 +27,6 @@
 		public string hotcorner_custom_command { get; set; }
 		public string[] dock_names { get; set; }
 
-		public WindowOverviewType window_overview_type { get; set; }
-
-		public ActionType hotcorner_topleft { get; set; }
-		public ActionType hotcorner_topright { get; set; }
-		public ActionType hotcorner_bottomleft { get; set; }
-		public ActionType hotcorner_bottomright { get; set; }
-
 		static BehaviorSettings? instance = null;
 
 		private BehaviorSettings ()
--- src/gala/src/PluginManager.vala	2015-11-07 02:44:07.863037212 +0900
+++ src/gala/src/PluginManager.vala.qq	2015-11-07 02:49:19.998658942 +0900
@@ -108,10 +108,6 @@
 				return false;
 			}
 
-			if (!check_provides (info.name, info.provides)) {
-				return false;
-			}
-
 			info.module_name = plugin_name;
 			module.make_resident ();
 
--- src/gala/src/Main.vala	2015-11-07 02:44:07.862037245 +0900
+++ src/gala/src/Main.vala.qq	2015-11-07 02:47:45.447802690 +0900
@@ -38,7 +38,7 @@
 		    Meta.exit (Meta.ExitCode.ERROR);
 		}
 
-		Meta.Plugin.manager_set_plugin_type (new WindowManagerGala ().get_type ());
+		//Meta.Plugin.manager_set_plugin_type (new WindowManagerGala ().get_type ());
 
 		//Meta.set_wm_name ("Mutter(Gala)");
 
--- src/gala/vapi/Makefile.am	2015-11-07 03:00:43.925131000 +0900
+++ src/gala/vapi/Makefile.am.qq	2015-11-07 03:15:08.768163548 +0900
@@ -4,12 +4,10 @@
 	libbamf3.vapi \
 	gdesktopenums-3.0.vapi \
 	gnome-desktop-3.0.vapi \
-	libmutter.vapi \
 	xfixes-4.0.vapi \
 	$(NULL)
 
 DEPS = \
-	libmutter.deps \
 	gnome-desktop-3.0.deps \
 	$(NULL)
 
--- src/gala/vapi/Makefile.am	2015-11-07 03:19:56.616592810 +0900
+++ src/gala/vapi/Makefile.am.qq	2015-11-07 03:28:20.619835088 +0900
@@ -3,12 +3,10 @@
 	config.vapi \
 	libbamf3.vapi \
 	gdesktopenums-3.0.vapi \
-	gnome-desktop-3.0.vapi \
 	xfixes-4.0.vapi \
 	$(NULL)
 
 DEPS = \
-	gnome-desktop-3.0.deps \
 	$(NULL)
 
 EXTRA_DIST = \
--- src/gala/vapi/Makefile.am	2015-11-07 03:35:40.882196702 +0900
+++ src/gala/vapi/Makefile.am.qq	2015-11-07 23:19:18.056852254 +0900
@@ -2,8 +2,6 @@
 	cogl-fixes.vapi \
 	config.vapi \
 	libbamf3.vapi \
-	gdesktopenums-3.0.vapi \
-	xfixes-4.0.vapi \
 	$(NULL)
 
 DEPS = \
--- src/gala/src/DBus.vala	2015-11-07 23:23:21.155769401 +0900
+++ src/gala/src/DBus.vala.qq	2015-11-07 23:32:42.083119001 +0900
@@ -21,12 +21,12 @@
 	public class DBus
 	{
 		static DBus? instance;
-		static WindowManager wm;
+		//static WindowManager wm;
 
 		[DBus (visible = false)]
-		public static void init (WindowManager _wm)
+		public static void init ()
 		{
-			wm = _wm;
+		//	wm = _wm;
 
 			Bus.own_name (BusType.SESSION, "org.pantheon.gala", BusNameOwnerFlags.NONE,
 				(connection) => {
@@ -39,9 +39,9 @@
 				},
 				() => {},
 				() => warning ("Could not acquire name\n") );
 		}
 
-		private DBus ()
+		/* private DBus ()
 		{
 			if (wm.background_group != null)
 #if HAS_MUTTER314
@@ -51,11 +51,11 @@
 #endif
 			else
 				assert_not_reached ();
-		}
+		} */
 
 		public void perform_action (ActionType type)
 		{
-			wm.perform_action (type);
+			//wm.perform_action (type);
 		}
 
 		const double SATURATION_WEIGHT = 1.5;
@@ -101,7 +101,7 @@
 		 * @param reference_width  Width of the rectangle
 		 * @param reference_height Height of the rectangle
 		 */
-		public async ColorInformation get_background_color_information (int monitor,
+		/* public async ColorInformation get_background_color_information (int monitor,
 			int reference_x, int reference_y, int reference_width, int reference_height)
 			throws DBusError
 		{
@@ -224,6 +224,6 @@
 			yield;
 
 			return { rTotal, gTotal, bTotal, mean, variance };
-		}
+		} */
 	}
 }
--- src/gala/src/DBus.vala	2015-11-07 23:56:59.222670227 +0900
+++ src/gala/src/DBus.vala.qq	2015-11-07 23:58:41.773260499 +0900
@@ -53,10 +53,10 @@
 				assert_not_reached ();
 		} */
 
-		public void perform_action (ActionType type)
+		/* public void perform_action (ActionType type)
 		{
-			//wm.perform_action (type);
-		}
+			wm.perform_action (type);
+		} */
 
 		const double SATURATION_WEIGHT = 1.5;
 		const double WEIGHT_THRESHOLD = 1.0;
