--- src/properties-cpp/doc/Doxyfile.in~	2015-04-11 01:08:37.453370000 +0900
+++ src/properties-cpp/doc/Doxyfile.in	2015-04-11 01:22:25.952871516 +0900
@@ -929,7 +929,7 @@
 # robust against future updates. Doxygen will copy the style sheet file to
 # the output directory.
 
-HTML_EXTRA_STYLESHEET  = @CMAKE_CURRENT_SOURCE_DIR@/extra.css
+HTML_EXTRA_STYLESHEET  = 
 
 # The HTML_EXTRA_FILES tag can be used to specify one or more extra images or
 # other source files which should be copied to the HTML output directory. Note
