--- src/bamf/src/bamf-matcher.c~	2014-08-11 04:11:29.725696000 +0900
+++ src/bamf/src/bamf-matcher.c	2014-08-11 04:13:27.520340307 +0900
@@ -29,7 +29,7 @@
 
 #include <strings.h>
 
-#define BAMF_INDEX_NAME "bamf-2.index"
+#define BAMF_INDEX_NAME "/var/cache/bamf-2.index"
 
 G_DEFINE_TYPE (BamfMatcher, bamf_matcher, BAMF_DBUS_TYPE_MATCHER_SKELETON);
 #define BAMF_MATCHER_GET_PRIVATE(obj) (G_TYPE_INSTANCE_GET_PRIVATE(obj, \
@@ -1361,7 +1361,7 @@
 
       bamf_matcher_add_new_monitored_directory (self, directory);
 
-      bamf_file = g_build_filename (directory, BAMF_INDEX_NAME, NULL);
+      bamf_file = g_build_filename (BAMF_INDEX_NAME, NULL);
 
       if (g_file_test (bamf_file, G_FILE_TEST_EXISTS))
         {
