--- src/bamf/src/bamf-matcher.c~	2014-08-10 20:34:36.474700000 +0900
+++ src/bamf/src/bamf-matcher.c	2014-08-10 23:05:43.569588181 +0900
@@ -1102,6 +1102,8 @@
   GList *dirs = NULL;
   char *path;
 
+  dirs = g_list_prepend (dirs, g_strdup ("/var/cache/bamf"));
+
   dirs = list_prepend_desktop_file_env_directories (dirs, "XDG_DATA_DIRS");
 
   if (!g_list_find_custom (dirs, "/usr/share/applications", (GCompareFunc) g_strcmp0))
