--- src/bamf/src/bamf-matcher.c~	2014-08-11 00:47:54.505809427 +0900
+++ src/bamf/src/bamf-matcher.c	2014-08-11 00:47:14.103555000 +0900
@@ -1128,5 +1135,8 @@
   /* Include also the user desktop folder, but without its subfolders */
   dirs = g_list_prepend (dirs, g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP)));
 
+  /* Include also the cache directory */
+  dirs = g_list_prepend (dirs, g_strdup ("/var/cache/bamf"));
+
   return dirs;
 }
 
