--- /etc/dbus-1/system.d/org.freedesktop.login1.conf	2014-03-04 05:06:29.000000000 +0900
+++ /etc/dbus-1/system.d/org.freedesktop.login1.conf	2014-06-13 11:58:21.184546991 +0900
@@ -82,6 +82,10 @@
 
                 <allow send_destination="org.freedesktop.login1"
                        send_interface="org.freedesktop.login1.Manager"
+                       send_member="KillUser"/>
+
+                <allow send_destination="org.freedesktop.login1"
+                       send_interface="org.freedesktop.login1.Manager"
                        send_member="ActivateSession"/>
 
                 <allow send_destination="org.freedesktop.login1"
