--- src/switchboard-plug-sharing/src/Widgets/BluetoothPage.vala~	2017-05-16 08:48:54.210178706 +0900
+++ src/switchboard-plug-sharing/src/Widgets/BluetoothPage.vala	2017-05-16 08:57:40.050173161 +0900
@@ -33,9 +33,6 @@
         bluetooth_settings = new GLib.Settings ("org.pantheon.desktop.wingpanel.indicators.bluetooth");
         sharing_settings = new GLib.Settings ("org.gnome.desktop.file-sharing");
 
-        sharing_settings.bind ("bluetooth-obexpush-enabled", service_switch, "active", SettingsBindFlags.NO_SENSITIVITY);
-        sharing_settings.bind ("bluetooth-accept-files", accept_combo, "active-id", SettingsBindFlags.DEFAULT);
-        sharing_settings.bind ("bluetooth-notify", notify_switch, "active", SettingsBindFlags.DEFAULT);
 
         service_switch.notify ["active"].connect (() => {
             set_service_state ();
@@ -81,7 +81,7 @@
 
     private void set_service_state () {
         if (bluetooth_settings.get_boolean ("bluetooth-enabled")) {
-            update_state (sharing_settings.get_boolean ("bluetooth-obexpush-enabled") ? ServiceState.ENABLED : ServiceState.DISABLED);
+            update_state (ServiceState.NOT_AVAILABLE);
         } else {
             update_state (ServiceState.NOT_AVAILABLE);
         }
