--- src/bamf/src/bamf-matcher.c~	2014-08-11 05:15:01.355611000 +0900
+++ src/bamf/src/bamf-matcher.c	2014-08-11 05:20:32.359545051 +0900
@@ -1101,6 +1101,7 @@
 {
   GList *dirs = NULL;
   char *path;
+  char *cache;
 
   dirs = list_prepend_desktop_file_env_directories (dirs, "XDG_DATA_DIRS");
 
@@ -1126,6 +1127,11 @@
   /* Include also the user desktop folder, but without its subfolders */
   dirs = g_list_prepend (dirs, g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP)));
 
+  /* Include the cache folder */
+  cache = g_build_filename ("/var/cache/bamf", NULL);
+  dirs = g_list_prepend (dirs, cache);
+  g_free (cache);
+
   return dirs;
 }
 
